### PR TITLE
[GLUTEN-4159][BUILD] link to target ZLIB before add velox library

### PR DESCRIPTION
**Pull request by dcoliversun**
## What changes were proposed in this pull request?

This PR aims to link to target ZLIB before add velox library to fix error in `/dev/package.sh` script.
(Fixes: \#4159)

## How was this patch tested?

Add new test.

### CODE REVIEW

**Comment by dcoliversun** [comment]
cc @zhouyuan Could you please check this error?  I can reproduce this error on my machine
```plain
-- Found Boost: /usr/local/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found suitable version "1.72.0", minimum required is "1.51.0") found components: context filesystem program_options regex system thread 
-- Found folly: /usr/local
CMake Error at velox/CMakeLists.txt:68 (message):
  Velox library not exists:
  /opt/gluten/ep/build-velox/build/velox_ep/_build/release/velox/exec/tests/utils/libvelox_exec_test_lib.a
Call Stack (most recent call first):
  velox/CMakeLists.txt:125 (add_velox_dependency)
  velox/CMakeLists.txt:348 (add_velox_dependencies)
```

**Comment by surnaik** [comment]
@zhouyuan @dcoliversun 

Velox removed the VELOX_BUILD_BENCHMARKS flag recently, which is causing this 

https://github.com/facebookincubator/velox/commit/f2cdf813be50cc165a104b93cabbb56319bd4822

**Comment by dcoliversun** [comment]
@surnaik Thanks for reply. @PHILO-HE is working on this :)https://github.com/oap-project/gluten/issues/4159#issuecomment-1869197230

File: .github/workflows/velox_be.yml [hunk 1]
```diff
@@ -560,3 +560,23 @@ jobs:
         if: ${{ always() }}
         run: |
           docker stop static-build-test-$GITHUB_RUN_ID || true
+  static-package-test:
```
Comment (PHILO-HE) [inline .github/workflows/velox_be.yml:563]: Maybe, better to name it like `build-script-test`. We can enrich the tests in this CI job in the future. Thanks!

Comment (dcoliversun) [reply .github/workflows/velox_be.yml]: have done :)

File: .github/workflows/velox_be.yml [hunk 2]
```diff
@@ -560,3 +560,23 @@ jobs:
...
+        run: |
+          docker exec build-script-test-$GITHUB_RUN_ID bash -c '
+          source /env.sh && \
+          sudo yum -y install patch && \
+          cd /opt/gluten/ && \
+          ./dev/package.sh'
+      - name: Exit docker container
+        if: ${{ always() }}
+        run: |
+          docker stop build-script-test-$GITHUB_RUN_ID || true
```
Comment (PHILO-HE) [inline .github/workflows/velox_be.yml:582]: Nit: let's leave one empty line at the end. Thanks!

Comment (dcoliversun) [reply .github/workflows/velox_be.yml]: thanks, updated

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**